diff --git a/uloop.c b/uloop.c
index ea160a0..8343bc5 100644
--- a/uloop.c
+++ b/uloop.c
@@ -256,7 +256,7 @@ static int uloop_fetch_events(int timeout)
 {
 	int n, nfds;
 
-	nfds = epoll_wait(poll_fd, events, ARRAY_SIZE(events), timeout);
+	nfds = epoll_wait(poll_fd, events, ARRAY_SIZE(events), timeout < 0 ? 8000 : timeout);
 	for (n = 0; n < nfds; ++n) {
 		struct uloop_fd_event *cur = &cur_fds[n];
 		struct uloop_fd *u = events[n].data.ptr;
